### PR TITLE
RaftDriver: Add `assert_detail`

### PR DESCRIPTION
**Pull request by achamayou**
Spun out of #5973, convenient in retirement scenarios.

### CODE REVIEW

File: tests/raft_scenarios/retire_one [hunk 1]
```diff
@@ -58,7 +58,8 @@ dispatch_all
 
 assert_is_primary,0
 assert_is_backup,1
```
Comment (eddyashton) [inline tests/raft_scenarios/retire_one:60]: We could also replace these assert lines (`assert_detail,1,leadership_state,Backup`), if we want to consolidate actions? I don't know if its helpful though.

**Review by eddyashton — APPROVED** [review]

File: tests/raft_scenarios/retire_one (hunk 1 above)
Comment (achamayou) [reply tests/raft_scenarios/retire_one:60]: I started doing that, and then noticed the `assert_isnot_*` variants, which can't easily be replaced with `assert_detail`. But now that you mention it, we could easily `assert_!detail`.

File: src/consensus/aft/test/driver.h [hunk 2]
```diff
@@ -1155,4 +1095,46 @@ class RaftDriver
...
+        "  Note over {}: Node does not have detail {}", node_id, detail);
+      throw std::runtime_error(fmt::format(
+        "Node {} does not have detail {} on line {}",
+        node_id,
+        detail,
+        std::to_string((int)lineno)));
+    }
+
+    std::string value = d[detail];
+    if (equal ? value != expected : value == expected)
```
Comment (eddyashton) [inline src/consensus/aft/test/driver.h:1120]:
```suggestion
    if (equal ? (value != expected) : (value == expected))
```

Nit: Odd place for a ternary, so I think this is a big help to readability.

Comment (achamayou) [reply src/consensus/aft/test/driver.h:1120]: Ok as a one-off, but otherwise the clang-format is the law.

**Pull request merged**